### DISCUSSION TITLE
Does Statiq Dev support F#?

### DISCUSSION THREAD

**Discussion by sonicbhoc** [statiqdev/Discussions#141] [Q&A] [answered]
The documentation says it can generate API documentation from .NET code, but the default globbing pattern only matches .cs files.

Since C# isn't the only .NET language, I figured I'd ask before I sink a lot of development effort into evaluating this tool for our use. Thanks!

**Comment by daveaglick** [comment, marked as answer]
Good question. Unfortunately I think the short answer is no, and to do so would require more knowledge of F# than I poses.

The longer answer is that while the [Buildalyzer](https://github.com/daveaglick/Buildalyzer) tool Statiq uses to process project files and figure out what source files constitute a project, most of the rest of the process is based on Roslyn. That includes the `AnalyzeCSharp` module in Statiq that processes the Roslyn syntax tree and outputs documents for each symbol (which is what drives documentation generation at the theme level). So in theory, Statiq could probably be adapted to VB code without too much trouble since Roslyn can also generate syntax trees for that language, but a `AnalyzeFSharp` module would require a complete from-scratch methodology for generating a syntax tree from F# code and outputting corresponding Statiq documents for it. That's not to say it couldn't be done - it's just way outside my extremely limited knowledge and ability with F# to do.

**Reply by sonicbhoc** [reply]
Thank you! Maybe once I'm more knowledgeable I can take a crack at it.

Unfortunately, our organization is starting to do F# code now, so as cool as Statiq Docs is, I'll probably just stick to fsdocs instead.

**Reply by daveaglick** [reply]
I’m a fan of F# generally and it’s cool your org is leaning in! Unfortunately just don’t have enough hours in the day to train myself up on it beyond some hello world stuff.
